airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -912,9 +912,9 @@ protected void verifyCheckpointStatesBySeconds() throws Exception {
 
     assertEquals(recordsToCreate, extractRecordMessages(dataFromSecondBatch).size());
     final List<AirbyteStateMessage> stateMessagesCDC = extractStateMessages(dataFromSecondBatch);
-    if (!isOnLegacyPostgres()) {
-      assertTrue(stateMessagesCDC.size() > 1, "Generated only the final state.");
-    }
+    // We expect only one cdc state message, as all the records are inserted in a single transaction. Since
```

Comment:
thank you for the additional comment. Does that mean that somehow, your change is also fixing legacy postgres?

Reply:
Not quite. For legacy postgres, only one state message was expected (not sure why). 

So this change just streamlines the test to check for just one state message instead of multiple (which I'm not sure was a good test to begin with)

Reply:
we NEVER are on legacy postgres... Let's remove that method altogether, no?